Dangling reference in `parse_scenario()`

Trying to compile libcosim with GCC 14, I get the following error message:
```
[...]/src/cosim/scenario_parser.cpp: In function ‘cosim::scenario::scenario cosim::parse_scenario(const std::filesystem::__cxx11::path&, const std::unordered_map<int, manipulable*>&)’:
[...]/src/cosim/scenario_parser.cpp:214:21: error: possibly dangling reference to a temporary [-Werror=dangling-reference]
  214 |         const auto& [index, simulator] =
      |                     ^~~~~~~~~~~~~~~~~~
[...]/src/cosim/scenario_parser.cpp:215:27: note: the temporary was destroyed at the end of the full expression ‘cosim::{anonymous}::find_simulator((* & simulators), cosim::{anonymous}::specified_or_default(const YAML::Node&, const std::string&, std::optional<std::__cxx11::basic_string<char> >)(std::__cxx11::basic_string<char>(((const char*)"model"), std::allocator<char>()), std::optional<std::__cxx11::basic_string<char> >(defaultOpts.cosim::{anonymous}::defaults::model)))’
  215 |             find_simulator(simulators, specified_or_default(event, "model", defaultOpts.model));
      |             ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
In other words, the problem is that `find_simulator()` returns by value, not by reference, but we still try to capture the result by reference.

The fix is simple: Just drop the `&` in line 214.

--- a/src/cosim/scenario_parser.cpp
+++ b/src/cosim/scenario_parser.cpp
@@ -211,7 +211,7 @@ scenario::scenario parse_scenario(
         auto triggerTime = trigger.as<double>();
         auto time = to_time_point(triggerTime);
 
-        const auto& [index, simulator] =
+        const auto [index, simulator] =
             find_simulator(simulators, specified_or_default(event, "model", defaultOpts.model));
         auto varName =
             specified_or_default(event, "variable", defaultOpts.variable);

